CMakeLists.txt

```diff
@@ -333,22 +340,6 @@ endif()
 
 target_link_libraries(avif_obj PRIVATE avif_enable_warnings)
 
-if(AVIF_ENABLE_NODISCARD)
-    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.21.0)
-        set(CMAKE_C_STANDARD 23)
-        set_property(TARGET avif_obj PROPERTY C_STANDARD 23)
-    else()
```

Comment:
If we redefine `AVIF_ENABLE_NODISCARD` to be a request rather than a requirement, we don't need this `else` block.